### PR TITLE
Add auto-reapproval on identical force-push

### PR DESCRIPTION
**Pull request by matthauck**

### CODE REVIEW

**Comment by matthauck** [comment]
btw. doing some manual testing right now and it's not quite working yet... :thinking:

**Comment by trevershick** [comment]
So I was reviewing this and I didn't see anything wrong, but then you added a commit and said `Make it work for real` which then made me question my existence for being since I couldn't tell it wouldn't have worked.  Great.  Existential angst due to a PR.  At any rate, I don't see anything glaringly bad/wrong but the logic is pretty rough to follow and I think I would decompose it so that the workflow portion is more readable statements vs actual bit comparisons `is_approved(review)` vs `review.state === 'approved'`.  This is just the first one I can think of.  At any rate, I can't 'see' anything wrong with it.  Of course this is all based on it working...

File: tests/force_push_test.rs [hunk 1]
```diff
@@ -477,26 +481,33 @@ fn test_force_push_identical_wrong_previous_approval() {
...
+    let before_hash = "abcdef0999999";
+    let after_hash = "1111abc9999999";
+
+    github.mock_get_statuses("some-user", "some-repo", before_hash, Ok(vec![]));
     github.mock_get_timeline(
         "some-user",
         "some-repo",
         32,
         Ok(vec![
+            github::TimelineEvent::new_review(
```
Comment (matthauck) [inline tests/force_push_test.rs]: realized this test was not testing what I thought it was. I messed up `found_dismiss` doing this refactor and this test didn't catch it. Made sure this time it did fail before moving `found_dismiss = true;` back to its proper spot.

**Comment by trevershick** [comment]
`multiple if let(...) => tuples` I think.
 :) . 
although that can get ugly quick if you have long var names or you're calling functions.

```
if let (Some(a), Some(b)) = (expr X, expr Y)
```

It looks better.  If it's working :shipit:

**Comment by matthauck** [comment]
cool, thanks. mind re-approving?

**Pull request merged**